src/prefect/infrastructure/provisioners/cloud_run.py

```diff
@@ -0,0 +1,237 @@
+                        f" {tmpdir}/prefect-cloud-run-key.json"
+                        f" --iam-account=prefect-cloud-run@{self._project}.iam.gserviceaccount.com"
+                    )
+                )
+            except subprocess.CalledProcessError as e:
+                raise RuntimeError(
+                    "Error creating service account key. Please ensure you have the"
+                    " necessary permissions."
+                ) from e
+            key = json.loads((Path(tmpdir) / "prefect-cloud-run-key.json").read_text())
+        return key
+
+    async def _assign_roles(self):
+        try:
+            await self._run_command(
+                shlex.split(
+                    "gcloud projects add-iam-policy-binding"
+                    f' {self._project} --member="serviceAccount:prefect-cloud-run@{self._project}.iam.gserviceaccount.com"'
+                    ' --role="roles/iam.serviceAccountUser"'
+                )
+            )
+            await self._run_command(
+                shlex.split(
+                    "gcloud projects add-iam-policy-binding"
+                    f' {self._project} --member="serviceAccount:prefect-cloud-run@{self._project}.iam.gserviceaccount.com"'
+                    ' --role="roles/run.developer"'
+                )
+            )
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(
+                "Error assigning roles to service account. Please ensure you have the"
+                " necessary permissions."
+            ) from e
+
+    async def _create_gcp_credentials_block(
+        self, work_pool_name: str, key: dict, client: PrefectClient
+    ) -> UUID:
+        credentials_block_type = await client.read_block_type_by_slug("gcp-credentials")
+
+        credentials_block_schema = (
+            await client.get_most_recent_block_schema_for_block_type(
+                block_type_id=credentials_block_type.id
+            )
+        )
+
+        try:
+            block_doc = await client.create_block_document(
+                block_document=BlockDocumentCreate(
+                    name=f"{work_pool_name}-push-pool-credentials",
+                    data={"service_account_info": key},
+                    block_type_id=credentials_block_type.id,
+                    block_schema_id=credentials_block_schema.id,
+                )
+            )
+            return block_doc.id
+        except ObjectAlreadyExists:
+            block_doc = await client.read_block_document_by_name(
+                name=f"{work_pool_name}-push-pool-credentials",
+                block_type_slug="gcp-credentials",
+            )
+            return block_doc.id
+
+    async def provision(self, work_pool_name: str, base_job_template: dict):
+        async with get_client() as client:
+            await self._check_for_gcloud()
+            self._project = await self._get_current_project()
+            self._region = await self._get_default_region()
```

Comment:
Should we add a check to ensure we can authenticate, or will that error surface when retrieving the current project?

Reply:
I added a check to make sure there's at least one authenticated project.